Scripts/RemediationScripts/Remediate-PTAControls.ps1

```diff
@@ -0,0 +1,915 @@
+    
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Subscription Name: $($context.Subscription.Name)"
+    Write-Host "Subscription ID: $($context.Subscription.SubscriptionId)"
+    Write-Host "Account Name: $($context.Account.Id)"
+    Write-Host "Account Type: $($context.Account.Type)"
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "Checking if $($context.Account.Id) is allowed to run this script..."
+
+    # Checking if the current account type is "User"
+    if ($context.Account.Type -ne "User")
+    {
+        Write-Host "WARNING: This script can only be run by `User` Account Type. Account Type of $($context.Account.Id) is: $($context.Account.Type)" -ForegroundColor $([Constants]::MessageType.Warning)
+        break
+    }
+
+    Write-Host "*** To Create PIM role assignment(s) with SC-ALT account for critical permanent role assignment(s) and remove critical permanent role assignment(s) for which PIM role assignment(s) is successfully created in the Subscription or in resource group, Owner and higher privileges on the subscription are required. ***" -ForegroundColor $([Constants]::MessageType.Info)
+    
+    # Safe Check: Current user need to be either UAA or Owner for the subscription
+    write-Host "Checking if the current user has required role..."
+    $currentLoginRoleAssignments = Get-AzRoleAssignment -SignInName $context.Account.Id -Scope "/subscriptions/$($SubscriptionId)";
+
+    $requiredRoleDefinitionName = @("Owner", "User Access Administrator")
+    if(($currentLoginRoleAssignments | Where { $_.RoleDefinitionName -in $requiredRoleDefinitionName} | Measure-Object).Count -le 0 )
+    {
+        Write-Host "Warning: This script can only be run by an [$($requiredRoleDefinitionName -join ", ")]." -ForegroundColor Yellow
+        #return;
+    }
+    else
+    {
+        Write-Host "Current user [$($currentSub.Account.Id)] has the required permission for subscription [$($SubscriptionId)]." -ForegroundColor Green
+    }
+    
+    $currentUser = (Get-AzAdUser -ObjectId $context.Account.Id).Id
+    #List of role assignment details 
+    $roleAssignmentDetails = @()
+
+    Write-Host "checking if subscription contains one or more service adminstrator role assignment(s)"...
+
+    $roleAssignmentDetails = Get-AzRoleAssignment -scope "/subscriptions/$($SubscriptionId)"
+    $serviceAdminstrator = $roleAssignmentDetails | Where-Object {$_.Scope -eq "/subscriptions/$($SubscriptionId)" -and ($_.RoleDefinitionName -split";" -contains "ServiceAdministrator" -or $_.RoleDefinitionName -split";" -contains "CoAdministrator")}
+    
+    # Checking the count of service adminstrator role assignment in the subscription.
+    if(($serviceAdminstrator |Measure-Object).count -lt 1)
+    {
+        Write-Host "WARNING: This script can only be run when the subscription contains one or more service admin role assignment(s)." -ForegroundColor $([Constants]::MessageType.Warning)
```

Comment:
Please rephrase this as - "WARNING: This script can only be run when there is a Service Administrator configured for the Subscription."